Added "keep" anonymizer

paulo-raca opened the pull request:
## Change Description

This adds `keep` anonymizer. This is a no-op and doesn't modify the input string.

Having it allows us to keep track of all PII data in the anonymizer output, even when we want to preserve some types of PII.

### Example
I'm making a demo that highlights all PIIs and anonymizes only phone numbers.

When using `anonymizers: {"DEFAULT":{"type":"keep"}, "PHONE_NUMBER": {"type":"replace"}}`

> I'm John Smith and my phone is +1234567890

Becomes

> I'm `John Smith` and my phone is `<PHONE_NUMBER>`

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [ ] I have signed the CLA (if required)
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

omri374 commented:
Thanks for this great contribution! I've thing I'm curious about is what happens when we have overlaps. Currently the package resolves overlaps using a set of rules. It is possible that the output would not be the same as the input in case of overlaps. Have you looked into this by any chance? Consider adding an integration test for this if it is really an issue.

paulo-raca commented:
> One thing I'm curious about is what happens when we have overlaps. 

I haven't looked into it yet, but I think having a no-op anonymizer instead of ignoring the entity type would help keeping th behavior consistent.

> Consider adding an integration test for this if it is really an issue

I'll look into this.

paulo-raca commented:
@omri374, I've added e2e tests and the summary is that `I'm George Washington Square Park` becomes `I'm George WashingtonWashington Square Park`.

It is a bit weird, but it is consistent with the Partial intersection rules described in the docs, so I'm good with it :+1: 

I also installed pre-commit and fixed the linting errors

paulo-raca commented:
/azp run

omri374 reviewed (dismissed):
LGTM, Thanks!

omri374 commented:
/azp run

paulo-raca commented:
@microsoft-github-policy-service agree

dorlugasigal commented:
this looks great @paulo-raca
can you modify the tests name to have a `given_when_then` or `given_then` test names
and an additional sample notebook under `docs/sample/python`?

paulo-raca commented:
>can you modify the tests name to have a `given_when_then` or `given_then` test names

Whoops, looks a copy-pasting mistake :sweat_smile: 

> and an additional sample notebook under `docs/sample/python`?

Added, let me know if it is enough :+1:

dorlugasigal reviewed (approved):
lgtm

omri374 commented:
/azp run
